### ISSUE TITLE
Rust code generation for TypeScript does not include type for both data and variables

### ISSUE THREAD

**Issue by kesne** [facebook/relay#3648]
When using the TypeScript language extension for the JS compiler, it outputs a unified type that contains both the response data type and the variable type, which is useful for libraries to consume when typing hooks. The currently typescript code generation only outputs separate types, one for the variables, and one for the data.

For example, I'd expect something like this to get output to the generated file when defining an `ExampleQuery`:
```typescript
export type ExampleQuery = {
    readonly data: ExampleQuery$data;
    readonly variables: ExampleQuery$variables;
};
```

**Comment by maraisr** [comment]
Isnt it already doing this?

```
export type pagePressQuery = {
  variables: pagePressQueryVariables;
  response: pagePressQueryResponse;
};
```

**Comment by kesne** [comment]
@maraisr I'm only seeing that get generated in the JS compiler, not the rust compiler.

**Comment by maraisr** [comment]
That copy paste was from a rust compiler output from my current project 👀

**Comment by kesne** [comment]
That's super odd, I don't see that at all. On my project when running the latest rust compiler, I don't see that type generated. The variables naming convention is also different: I see `data` used instead of `response`, and the types are `$variables` and `$data` postfixed instead of the camel case example you posted.

**Comment by maraisr** [comment]
If you head on over to https://github.com/maraisr/meros/tree/main/examples/relay-with-helix — which is an OSS rust compiler enabled repo. Granted should probably upgrade it all to latest.

youll also find:

```ts
export type App_Query = {
	variables: App_QueryVariables;
	response: App_QueryResponse;
};
```

**Comment by maraisr** [comment]
Ah no!!! I found it during the updates — youre right.

In `0.0.0-main-2219c43b` you get types ✅
in `0.0.0-main-2d5d60fc` (current main) you dont get right types ❌

cc @kassens @josephsavona  this is a real issue, sorry for the re herrings all over the shop 😔

**Comment by wongmjane** [comment]
It appears this is due to the changes of the “Phase 3” typegen

For example, in 2710deff7b7aaa3c3c9944e84449662c5654fa36, there is a number of changes in the types, such as the removal of Query types even in Flow

**Comment by taras** [comment]
Just tried upgrading to relay-compiler 12 and found that generated ts files are missing `FragmentRefs` import. 

The imports look like this,

```ts
import { ReaderFragment } from "relay-runtime";
import {  } from "relay-runtime";
```

**Comment by wongmjane** [comment]
looks like it’s being fixed in a504352d32218112053716627c780616eac1700b

**Comment by maraisr** [comment]
💯 Just validated it as well, and yes that fixed it.
